domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2537,6 +2538,41 @@ public void select(String sql, Handler<AsyncResult<ResultSet>> replyHandler) {
     client.getConnection(conn -> select(conn, sql, closeAndHandleResult(conn, replyHandler)));
   }
 
+  static void queryAndAnalyze(SQLConnection conn, String sql, String statMethod,
+    Handler<AsyncResult<ResultSet>> replyHandler) {
+
+    long start = System.nanoTime();
+    conn.query(sql, res -> {
+      long queryTime = (System.nanoTime() - start);
+      StatsTracker.addStatElement(STATS_KEY + statMethod, queryTime);
+      if (res.failed()) {
+        log.error("queryAndAnalyze: " + res.cause().getMessage() + " - "
+          + sql, res.cause());
+        replyHandler.handle(Future.failedFuture(res.cause()));
+        return;
+      }
+      if (queryTime >= explainQueryThreshold * 1000000) {
+        final String explainQuery = "EXPLAIN ANALYZE " + sql;
+        conn.query(explainQuery, explain -> {
+          replyHandler.handle(res); // not before, so we have conn if it gets closed
+          if (explain.failed()) {
+            log.warn(explainQuery + " failed", explain.cause().getMessage());
+            return;
+          }
+          StringBuilder e = new StringBuilder();
+          for (JsonArray ar : explain.result().getResults()) {
+            if (ar.getValue(0) instanceof String) {
+              e.append("\n" + ar.getString(0));
+            }
+          }
+          log.warn(explainQuery + e.toString());
```

Comment:
explainQuery should be appended to e a few lines before. This avoids creating a temporary String needed for this + operation.